### PR TITLE
TEST/SOCKADDR: Full support for TCP

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6928/ for details (Mellanox internal link).

File: test/apps/sockaddr/sa_main.cc [hunk 1]
```diff
@@ -35,25 +39,85 @@ class application {
...
+        CONNECTION_CLIENT,
+        CONNECTION_SERVER
+    };
+
+    struct defaults {
+        static const int    CONN_COUNT;;
+        static const double CONN_RATIO;
+        static const size_t REQUEST_SIZE;
+        static const size_t RESPONSE_SIZE;
+        static const int    WAIT_TIME;
```
Comment (evgeny-leksikov) [inline test/apps/sockaddr/sa_main.cc]: not used?

File: test/apps/sockaddr/sa_main.cc [hunk 2]
```diff
@@ -35,25 +39,85 @@ class application {
...
+const size_t application::defaults::RESPONSE_SIZE = 1024;
+
 application::usage_exception::usage_exception(const std::string& message) :
                 error(message) {
 };
 
 application::application(int argc, char **argv) :
-                m_port(0) {
+                m_port(0),
+                m_conn_backlog(defaults::CONN_COUNT),
```
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply test/apps/sockaddr/sa_main.cc]: Usually, C++ solves it with corresponding default constructor + (optional) copy constructor of the same class/struct. (static instance is not really needed)
In your case, you need duplicate parameters in different classes/structures, I think it's more error friendly.
Anyway, it's too minor to block the PR, up to you.

File: test/apps/sockaddr/sa_tcp.h [hunk 3]
```diff
@@ -30,17 +30,33 @@ class tcp_connection : public connection {
 
     tcp_connection(int fd);
 
+    virtual void add_to_evpoll(evpoll_set& evpoll);
+
+    virtual size_t send(const char *buffer, size_t size);
```
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply test/apps/sockaddr/sa_tcp.h]: Ok for now

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6964/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov fixed

**Pull request merged**